core/src/runtime.ts

```diff
@@ -22,17 +23,20 @@ export interface RegisteredPlugin {
 
 export const createCapacitor = (win: WindowCapacitor): CapacitorInstance => {
   const cap: CapacitorInstance = win.Capacitor || ({} as any);
+  const capPlatforms: CapacitorPlatformsInstance = win.CapacitorPlatforms;
 
   const Plugins = (cap.Plugins = cap.Plugins || ({} as any));
 
   const webviewServerUrl =
     typeof win.WEBVIEW_SERVER_URL === 'string' ? win.WEBVIEW_SERVER_URL : '';
 
-  const getPlatform = () => getPlatformId(win);
+  const defaultGetPlatform = () => getPlatformId(win);
+  const getPlatform = capPlatforms.currentPlatform.getPlatform || defaultGetPlatform;
 
-  const isNativePlatform = () => getPlatformId(win) !== 'web';
+  const defaultIsNativePlatform = () => capPlatforms.currentPlatform.name !== 'web';
+  const isNativePlatform = capPlatforms.currentPlatform.isNativePlatform || defaultIsNativePlatform;
 
-  const isPluginAvailable = (pluginName: string): boolean => {
+  const defaultIsPluginAvailable = (pluginName: string): boolean => {
```

Comment:
Why does this have to get renamed?

Reply:
The defaults are not renaming the consts that are exposed, they are just holding the default version of that function so in the case of a custom platform not overriding a piece then the runtime will have the default to fall back on